### PR TITLE
[#117] Adds current page number and total pages

### PR DESCRIPTION
**Pull request by jborkowski**

### CODE REVIEW

File: src/main/java/pl/tomaszdziurko/jvm_bloggers/view/panels/CustomPageLinkModel.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.apache.wicket.markup.html.navigation.paging.IPageable;
+import org.apache.wicket.model.IModel;
+
+import java.io.Serializable;
+
+@RequiredArgsConstructor
+public class CustomPageLinkModel implements IModel<String>, Serializable {
+
+    protected final IPageable pageable;
+    protected final long pageNumber;
```
Comment (goostleek) [inline src/main/java/pl/tomaszdziurko/jvm_bloggers/view/panels/CustomPageLinkModel.java]: Why `protected` and not `private` instead?

File: src/main/java/pl/tomaszdziurko/jvm_bloggers/view/panels/CustomPageLinkModel.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import lombok.RequiredArgsConstructor;
+
+import org.apache.wicket.markup.html.navigation.paging.IPageable;
+import org.apache.wicket.model.IModel;
+
+import java.io.Serializable;
+
+@RequiredArgsConstructor
+public class CustomPageLinkModel implements IModel<String>, Serializable {
```
Comment (goostleek) [inline src/main/java/pl/tomaszdziurko/jvm_bloggers/view/panels/CustomPageLinkModel.java]: Redundant `Serializable` interface, `IModel` already implements it.

Comment (goostleek) [inline src/main/java/pl/tomaszdziurko/jvm_bloggers/view/panels/CustomPageLinkModel.java]: You can extend [`AbstractReadOnlyModel`](https://ci.apache.org/projects/wicket/apidocs/6.x/org/apache/wicket/model/AbstractReadOnlyModel.html) instead implementing `IModel` directly. This way you can get rid of empty [`setObject()`](https://github.com/tdziurko/jvm-bloggers/pull/167/files#diff-a2d7860c43e7bed9e9803f81433c6847R23) and [`detach()`](https://github.com/tdziurko/jvm-bloggers/pull/167/files#diff-a2d7860c43e7bed9e9803f81433c6847R23) method implementations.

File: src/main/java/pl/tomaszdziurko/jvm_bloggers/view/panels/CustomPageLinkModel.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @Override
+    public void setObject(String object) {
+    }
+
+    @Override
+    public void detach() {
+    }
+
```
Comment (goostleek) [inline src/main/java/pl/tomaszdziurko/jvm_bloggers/view/panels/CustomPageLinkModel.java]: Empty `detach()` and `setObject()` methods can be removed if you extend  [`AbstractReadOnlyModel`](https://ci.apache.org/projects/wicket/apidocs/6.x/org/apache/wicket/model/AbstractReadOnlyModel.html) instead implementing `IModel` directly.

File: src/main/java/pl/tomaszdziurko/jvm_bloggers/view/panels/CustomPageLinkModel.java (hunk 1 above)
Comment (jborkowski) [inline src/main/java/pl/tomaszdziurko/jvm_bloggers/view/panels/CustomPageLinkModel.java]: My oversight, this is residue from another approach.

File: src/main/java/pl/tomaszdziurko/jvm_bloggers/view/panels/CustomPageLinkModel.java (hunk 3 above)
Comment (jborkowski) [inline src/main/java/pl/tomaszdziurko/jvm_bloggers/view/panels/CustomPageLinkModel.java]: Done

**Comment by goostleek** [comment]
Coverage decreased due to additional code without tests. It would be nice if you provide a test for `CustomPageLinkModel` Wicket component. See [the documentation](https://ci.apache.org/projects/wicket/guide/8.x/guide/ajax.html) for approaches how to test Wicket components or existing [LoginPageSpec](https://github.com/tdziurko/jvm-bloggers/blob/ef35167ea62e5493b0135aa1f0a8f2db78d4b43c/src/test/groovy/pl/tomaszdziurko/jvm_bloggers/view/login/LoginPageSpec.groovy#L19) test.

File: src/main/java/pl/tomaszdziurko/jvm_bloggers/view/panels/CustomPageLinkModel.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
+package pl.tomaszdziurko.jvm_bloggers.view.panels;
+
+import lombok.RequiredArgsConstructor;
+
+import org.apache.wicket.markup.html.navigation.paging.IPageable;
+import org.apache.wicket.model.AbstractReadOnlyModel;
+
+@RequiredArgsConstructor
+public class CustomPageLinkModel extends AbstractReadOnlyModel<String> {
```
Comment (tdziurko) [inline src/main/java/pl/tomaszdziurko/jvm_bloggers/view/panels/CustomPageLinkModel.java]: This class looks really vague to me :) Can't we simply modify this css class in `LoopItem` generation code in `CustomPagingNavigator` and maybe extract this logic to separate method? Introducing model with so complicated logic (there are some magic `ifs` with no description that rewrite `idx` value for no clear reason).
It is very difficult to understand what is going on here.

File: src/test/groovy/pl/tomaszdziurko/jvm_bloggers/view/login/CustomPagingNavigatorSpec.groovy [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import org.apache.wicket.markup.repeater.data.DataView
+import org.apache.wicket.model.Model
+import org.apache.wicket.protocol.http.WebApplication
+import org.apache.wicket.util.tester.WicketTester
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.test.util.ReflectionTestUtils
+import pl.tomaszdziurko.jvm_bloggers.SpringContextAwareSpecification
+import pl.tomaszdziurko.jvm_bloggers.view.login.helpers.TestDataNavigationPage
+
+class CustomPagingNavigatorSpec extends SpringContextAwareSpecification {
```
Comment (goostleek) [inline src/test/groovy/pl/tomaszdziurko/jvm_bloggers/view/login/CustomPagingNavigatorSpec.groovy]: After introducing a new way of Wicket components testing in #182 there is no need to start-up entire Spring context. You can just extend [`MockSpringContextAwareSpecification`](https://github.com/tdziurko/jvm-bloggers/blob/master/src/test/groovy/pl/tomaszdziurko/jvm_bloggers/MockSpringContextAwareSpecification.groovy). See [`LoginPageSpec`](https://github.com/tdziurko/jvm-bloggers/blob/master/src/test/groovy/pl/tomaszdziurko/jvm_bloggers/view/login/LoginPageSpec.groovy) for example usage.

File: src/test/groovy/pl/tomaszdziurko/jvm_bloggers/view/login/helpers/TestDataNavigationPage.groovy [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+import org.apache.wicket.markup.html.WebPage
+import org.apache.wicket.markup.html.basic.Label
+import org.apache.wicket.markup.repeater.Item
+import org.apache.wicket.markup.repeater.data.DataView
+import org.apache.wicket.markup.repeater.data.ListDataProvider
+import pl.tomaszdziurko.jvm_bloggers.view.panels.CustomPagingNavigator
+/**
+ * Created by jborkowski on 28/03/16.
+ */
```
Comment (goostleek) [inline src/test/groovy/pl/tomaszdziurko/jvm_bloggers/view/login/helpers/TestDataNavigationPage.groovy]: According to coding guidelines test classes should not have any javadoc.
